File: Easy/lc1.py
# Two Sum
class Solution(object):
    def twoSum(self, nums, target):
        mp = {}

        for i in range(len(nums)):
            diff = target - nums[i]

            if diff in mp:
                return [mp[diff], i]

            mp[nums[i]] = i       
s=Solution()
print(s.twoSum([2,7,11,15],9))  # Output:  [0, 1]
print(s.twoSum([3,2,4],6))  # Output:   [1, 2]
print(s.twoSum([3,3],6))  # Output:     [0, 1]

File: Easy/lc1.java
import java.util.HashMap;
import java.util.Arrays;

Class Solution{
    public int[] twoSum(int[] nums, int target){
        HashMap <Integer , Integer> map = new HashMap<>();
        for(int i = 0;i<nums.length;i++){
            int diff = target - nums[i];
            if(map.containsKey(diff)){
                return new int[]{map.get(diff),i};
            }
            map.put(nums[i],i);
        }
        return new int[]{};
    }
    public static void main(String args[]){
        Solution sol = new Solution();
        int[] nums = {2,7,11,15};
        int target = 9;
        int[] result = sol.twoSum(nums,target);
        System.out.println(Arrays.toString(result));
    }
}